plugins/modules/docker_compose_v2.py

```diff
@@ -0,0 +1,371 @@
+  ansible.builtin.pip:
+    name: 
+    - python-on-whales
+    virtualenv: ~/ansible-compose-v2
+
+# Ensure project directory is present
+- name: Create directory if it does not exist
+  ansible.builtin.file:
+    path: uptime-kuma
+    state: directory
+# Create the docker compose file 
+- name: Create compose file with content
+  copy:
+    dest: "uptime-kuma/docker-compose.yaml"
+    content: |
+      version: '3.3'
+
+      services:
+        uptime-kuma:
+          image: louislam/uptime-kuma:1
+          container_name: uptime-kuma
+          volumes:
+            - ./uptime-kuma-data:/app/data
+          ports:
+            - 3001:3001
+          restart: always
+
+# Tear down any containers, remove images and volumes
+- name: Tear down all existing containers
+  vars:
+    ansible_python_interpreter: ansible-compose-v2/bin/python
+  community.docker.docker_compose_v2:
+    project_src: uptime-kuma/docker-compose.yaml
+    remove_volumes: true
+    remove_orphans: true
+    remove_images: all
+    state: absent
+
+# Pull the latest image and then start services
+- name: Pull image and start service
+  vars:
+    ansible_python_interpreter: ansible-compose-v2/bin/python
+  community.docker.docker_compose_v2:
+    project_src: uptime-kuma/docker-compose.yaml
+    pull: true
+    state: present
+
+# Restart services
+- name: Restart service
+  vars:
+    ansible_python_interpreter: ansible-compose-v2/bin/python
+  community.docker.docker_compose_v2:
+    project_src: uptime-kuma/docker-compose.yaml
+    restarted: true
+    state: present
+
+# Stop services
+- name: Stop service
+  vars:
+    ansible_python_interpreter: ansible-compose-v2/bin/python
+  community.docker.docker_compose_v2:
+    project_src: uptime-kuma/docker-compose.yaml
+    stopped: true
+    state: present
+
+# Start services, recreating all containers
+- name: Recreate service
+  vars:
+    ansible_python_interpreter: ansible-compose-v2/bin/python
+  community.docker.docker_compose_v2:
+    project_src: uptime-kuma/docker-compose.yaml
+    recreate: always
+    state: present
+'''
+
+RETURN = r'''
+# retuens are a bit finicky at the moment, see: https://github.com/gabrieldemarmiesse/python-on-whales/issues/395
+msg:
+    description: In docker failures, has docker command + some description. In other failures, stringified exception is returned. In success, descriptive message of what happened.
+    type: bool
+    returned: always
+    sample: 'true'
+changed:
+    description: If something was changed (currently, changed is (almost) always true as there is no check of current state).
+    type: bool
+    returned: always
+    sample: 'true'
+actions:
+    description: List stating which compose commands were run (not the entire command itself).
+    type: list
+    returned: success
+    sample: ["pull", "up"]
+rc:
+    description: Return code of executed docker command.
+    type: str
+    returned: failure from docker
+    sample: 12
+stderr_lines:
```

Comment:
Common return values should not be documented (https://docs.ansible.com/ansible/latest/reference_appendices/common_return_values.html#common-return-values).